### DISCUSSION TITLE
Sabbath Mode | Jewish Calendar Support

### DISCUSSION THREAD

**Discussion by RebbePod** [arendst/Tasmota#11857] [Ideas]
Add Hebrew Calendar support so one can make timers around Jewish holidays, when one is forbidden to use lights,plugs or any electronic devices. 
https://en.wikipedia.org/wiki/Sabbath_mode

There is a ready to use code:
https://github.com/yparitcher/libzmanim

It would be great if it can be added with an offset option (similar to sunset) so one can set a timer at x minutes before sabbath.

Lot's of modern appliances come with an option to activate a "sabbath mode" but unfortunately it has to be done manually, here the idea would be to have it automatically know when sabbath and the holidays are, as well as the proper starting times, which in the Jewish calendar a date begins at sunset (or a bit later). 

In addition, since during sabbath and holidays using electronic devices is prohibited having the ability to toggle a hardware button to not work during sabbath and holidays, will allow it that in case one unintentionally presses the switch / button, the lights won't go off etc.

Similar to: https://www.home-assistant.io/integrations/jewish_calendar/

**Comment by barbudor** [comment]
If you have home assistant or any home automation system, it would be easy to have it enabling a rule that will capture button press and avoid changing the light state.

**Reply by RebbePod** [reply]
Interesting idea.

Ideally I would like to implement it so it shouldn't be reliant on a home automation system. It shouldn't be to difficult considering there already is a timer function.

**Reply by Jason2866** [reply]
All Tasmota developers are doing the coding in there free time. So request from many users are on top of the "will be implemented if i have time for" list. This feature is first time asked.
It is not so easy and fast to add to Tasmota. Since Tasmota is Open Source you are welcome to do a PR or someone else is doing it.

**Reply by meingraham** [reply]
You may be able to implement this with Scripting. Tasmota Rules do not have a day of week variable. First, the "input" would need to be decoupled from the relay via SO73/SO114. The script would run when an action on the device is requested and invoke the `POWER ON` according to the day of the week.

**Reply by RebbePod** [reply]
Thanks for the replies, I fully understand it's only a request. I don't expect anything from anyone.

I'm not familiar with scripts I'll check it out.
Having just the day of the week won't be enough to calculate the holiday days, as well as the specific time the Sabbath and holiday starts.

**Reply by sfromis** [reply]
To get an automation following calendar rules (instead of only day/week cycle), I'd do it in a backend platform instead of each device. My preference would be Node-Red on the smarthome server where Mosquitto is running, but there are a number of other tools too.
